.github/workflows/build-system-test-react.yml

```diff
@@ -0,0 +1,138 @@
+# Description: this workflow runs build mega-apps that have all the connected components.
+
+name: Build System Test Canary - React
+
+on:
+  schedule:
+    - cron: '*/15 * * * *' # Run every 15 minutes
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: ci
+    strategy:
+      fail-fast: false
+      matrix:
+        framework: [react]
+        framework-version: [latest]
+        build-tool: [cra]
+        build-tool-version: [latest]
+        pkg-manager: [npm]
+        language: [ts]
+        node-version: [18]
```

Comment:
I think we can get rid of these since we're already using `include`.

Reply:
yea, we keep 1 here and add the rest 8 in the include. The reason I keep one here is just in case we can combine something together later so that we don't have to have a very long list in the matrix. 

I'll move this one to `include` if it turns out that we can't combine anything

Reply:
If I understand correctly, this will result in total of 8 configurations, not 9 because each `include` will overwrite the base matrix configuration, not append.

Reply:
> we can combine something together later we don't have to have a very long list in the matrix.

Speaking of this, what do you think about simplifying this to:

```yml
      matrix:
        framework: [react]
        framework-version: [latest]
        build-tool: [cra]
        build-tool-version: [latest]
        pkg-manager: [npm]
        language: [ts]
        node-version: [18]
        include:
          - node-version: 16 # change just the node-version
          - framework-version: 16 
          - node-version: 18
          - pkg-manager: yarn
             pkg-manager-version: 1.22.19
          # ...
```

so that this enforces the mental model of "base configuration" + "overrides"?

Reply:
I wish Github Workflow Matrix can be that smart 😓  [doc](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations)

For example the first include `- node-version: 16` will override it and turn to `{node-version: 16}` alone (similar as the "banana" in the official doc)

Reply:
> If I understand correctly, this will result in total of 8 configurations, not 9 because each include will overwrite the base matrix configuration, not append.

There are totally 9 of the mega-apps as listed in the description. And here's the latest build https://github.com/0618/amplify-ui/actions/runs/4682282681

Reply:
Yep, sorry about misdirection, understand the logic now!